lib/improver/tests/blending/weights/test_ChooseWeightsLinearFromDict.py

```diff
@@ -0,0 +1,513 @@
+                                      [[0.2, 0.2],
+                                       [0.2, 0.2]]]])
+
+        weighting_coord_name = "forecast_period"
+        config_coord_name = "model_configuration"
+        plugin = ChooseWeightsLinearFromDict(
+            weighting_coord_name, config_coord_name, self.config_dict)
+        new_weights_cube = plugin._create_new_weights_cube(cube, weights)
+        self.assertArrayAlmostEqual(expected_weights, new_weights_cube.data)
+        self.assertEqual(new_weights_cube.name(), "weights")
+
+    def test_alternative_name(self):
+        """Test a new weights cube is created as intended, with the desired
+        cube name when an alternative weights_coord_name is specified."""
+        cube = set_up_basic_model_config_cube()
+        weights = np.array([0., 0., 0.2])
+
+        expected_weights = np.array([[[[0., 0.],
+                                       [0., 0.]],
+                                      [[0., 0.],
+                                       [0., 0.]],
+                                      [[0.2, 0.2],
+                                       [0.2, 0.2]]]])
+        config_dict = self.config_dict
+        config_dict["uk_det"]["alternative_name"] = (
+            config_dict["uk_det"].pop("weights"))
+        weighting_coord_name = "forecast_period"
+        config_coord_name = "model_configuration"
+        plugin = ChooseWeightsLinearFromDict(
+            weighting_coord_name, config_coord_name, self.config_dict,
+            weights_coord_name="alternative_name")
+        new_weights_cube = plugin._create_new_weights_cube(cube, weights)
+        self.assertArrayAlmostEqual(expected_weights, new_weights_cube.data)
+        self.assertEqual(new_weights_cube.name(), "alternative_name")
+
+
+class Test__interpolate_to_create_weights(IrisTest):
+    """Test the _interpolate_to_create_weights method."""
+
+    def setUp(self):
+        """Set up configuration dictionary."""
+        self.config_dict = {"uk_det": {"forecast_period": [7, 12, 48, 54],
+                                       "weights": [0, 1, 1, 0],
+                                       "units": "hours"}}
+
+    def test_below_range(self):
+        """Test that interpolation works as intended when the forecast period
+        required for the interpolation output is below the range specified
+        within the inputs."""
+        cube = set_up_basic_model_config_cube()
+
+        expected_weights = np.array([[[[0., 0.],
+                                       [0., 0.]],
+                                      [[0., 0.],
+                                       [0., 0.]],
+                                      [[0.2, 0.2],
+                                       [0.2, 0.2]]]])
+
+        weighting_coord_name = "forecast_period"
+        config_coord_name = "model_configuration"
+        plugin = ChooseWeightsLinearFromDict(
+            weighting_coord_name, config_coord_name, self.config_dict)
```

Comment:
The above three lines are common to all three unit-tests in this class and should be moved to the setUp function. It then follows that setUp should only define self.plugin for the unit-tests to use.

Reply:
I've refactored this, as suggested.